src/main/java/io/confluent/copycat/hdfs/HdfsSinkTask.java

```diff
@@ -64,11 +65,18 @@ public void stop() throws CopycatException {
   public void put(Collection<SinkRecord> records) throws CopycatException {
     String topic;
     int partition;
-    for (SinkRecord record: records) {
+    for (SinkRecord record : records) {
       topic = record.getTopic();
       partition = record.getPartition();
       TopicPartition topicPart = new TopicPartition(topic, partition);
-      hdfsWriter.writeRecord(topicPart, record);
+      Map<TopicPartition, Long> offsets = hdfsWriter.getPreviousOffsets();
+      try {
+        if (!offsets.containsKey(topicPart) || record.getOffset() >= offsets.get(topicPart)) {
```

Comment:
So based on the test `record.getOffset() >= offsets.get(topicPart)`, the offsets returned by `getPreviousOffsets` is the first offset of the new file, not the last offset of the old file, right? Just want to make sure that is clear so we can avoid off-by-one errors.

Also, it'd be good to clearly document that, perhaps in the javadoc for `getPreviousOffsets`. The name doesn't make it entirely clear what it does -- before looking at the rest of the code, I thought it was going to be updated for every record written, but later found it refers to offsets that have actually been committed to their final location in HDFS.